synapse/config/registration.py

```diff
@@ -323,6 +324,11 @@ def generate_config_section(self, generate_secrets=False, **kwargs):
         # users cannot be auto-joined since they do not exist.
         #
         #autocreate_auto_join_rooms: true
+
+        # When auto_join_rooms is specified, setting this flag to False prevents
+        # guest accounts from being automatically joined to the rooms. Default
+        # true when auto_join_rooms is specified.
```

Comment:
does whether `auto_join_rooms` is set matter to the default?

Reply:
it doesn't do anything otherwise, similar to the auto create room setting above it.

Reply:
ok; I think it would be clearer just to omit the "when..." clause then.